apis/securesourcemanager/v1alpha1/securesourcemanagerrepository_types.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	refs "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var SecureSourceManagerRepositoryGVK = GroupVersion.WithKind("SecureSourceManagerRepository")
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// SecureSourceManagerRepositorySpec defines the desired state of SecureSourceManagerRepository
+// +kcc:proto=google.cloud.securesourcemanager.v1.Repository
+type SecureSourceManagerRepositorySpec struct {
+	/* Immutable. The Project that this resource belongs to. */
+	// +required
+	ProjectRef *refs.ProjectRef `json:"projectRef"`
+
+	/* Immutable. Location of the instance. */
+	Location string `json:"location"`
+
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ResourceID field is immutable"
+	// Immutable.
+	// The SecureSourceManagerRepository name. If not given, the metadata.name will be used.
+	ResourceID *string `json:"resourceID,omitempty"`
+
+	// Immutable. The name of the instance in which the repository is hosted, formatted as
+	// `projects/{project_number}/locations/{location_id}/instances/{instance_id}`
+	// +required
+	InstanceRef *SecureSourceManagerInstanceRef `json:"instanceRef,omitempty"`
```

Comment:
Question: Is SecureSourceManagerInstance currently supported in KCC? Do we plan to support it as well?

We can only reference to an external SecureSourceManagerInstance via "external" field until we support it in KCC.

Reply:
Yep it was added